homeassistant/components/ring/camera.py

```diff
@@ -136,6 +183,48 @@ async def handle_async_mjpeg_stream(
         finally:
             await stream.close()
 
+    async def async_handle_async_webrtc_offer(
+        self, offer_sdp: str, session_id: str, send_message: WebRTCSendMessage
+    ) -> None:
+        """Return the source of the stream."""
+
+        def message_wrapper(ring_message: RingWebRtcMessage) -> None:
+            if ring_message.error_code:
+                msg = ring_message.error_message or ""
+                send_message(WebRTCError(ring_message.error_code, msg))
+            elif ring_message.answer:
+                send_message(WebRTCAnswer(ring_message.answer))
+            elif ring_message.candidate:
+                send_message(
+                    WebRTCCandidate(
+                        RTCIceCandidateInit(
+                            ring_message.candidate,
+                            sdp_m_line_index=ring_message.sdp_m_line_index or 0,
+                        )
+                    )
+                )
+
+        return await self._device.generate_async_webrtc_stream(
+            offer_sdp, session_id, message_wrapper, keep_alive_timeout=None
+        )
+
+    async def async_on_webrtc_candidate(
+        self, session_id: str, candidate: RTCIceCandidateInit
+    ) -> None:
+        """Handle a WebRTC candidate."""
+        if candidate.sdp_m_line_index is None:
+            msg = "The sdp_m_line_index is required for ring webrtc streaming"
```

Comment:
We should raise here so the frontend gets notified, too, instead of only logging it.

Also be aware this method will be called for each candidate, which could result in many log entries for a single stream.

Reply:
This is updated to raise now which is then logged automatically.